librz/core/casm.c

```diff
@@ -64,6 +64,108 @@ RZ_API char *rz_core_asm_search(RzCore *core, const char *input) {
 	return ret;
 }
 
+static const char *has_esil(RzCore *core, const char *name) {
+	RzListIter *iter;
+	RzAnalysisPlugin *h;
+	rz_return_val_if_fail(core && core->analysis && name, NULL);
+	rz_list_foreach (core->analysis->plugins, iter, h) {
+		if (h->name && !strcmp(name, h->name)) {
+			return h->esil ? "Ae" : "A_";
+		}
+	}
+	return "__";
+}
+
+RZ_API RzCmdStatus rz_core_asm_plugins_print(RzCore *core, const char *arch, RzCmdStateOutput *state) {
+	int i;
+	const char *feat2, *feat;
+	RzAsm *a = core->rasm;
+	char bits[32];
+	RzAsmPlugin *h;
+	RzListIter *iter;
+	PJ *pj = 0;
+	if (arch) {
+		rz_list_foreach (a->plugins, iter, h) {
+			if (h->cpus && !strcmp(arch, h->name)) {
+				char *c = strdup(h->cpus);
+				int n = rz_str_split(c, ',');
+				for (i = 0; i < n; i++) {
+					rz_cons_println(rz_str_word_get0(c, i));
+				}
+				free(c);
+				break;
+			}
+		}
+	} else {
+		if (state->mode == RZ_OUTPUT_MODE_JSON) {
+			pj = state->d.pj;
+			rz_cmd_state_output_array_start(state);
```

Comment:
you can use this function even outside of the `if`. I actually suggest to use it, because it might be useful for other formats as well and that's the purpose of this function. To abstract a bit and avoid the caller to doing a lot of `if`..`else`...